testing/MLDB-1624-more-join-test.py

```diff
@@ -1251,34 +1251,27 @@ def test_equi_join(self):
             ]
         )     
 
-    @unittest.expectedFailure
+    # MLDB-1664
```

Comment:
@steadtler Good point for function calls.  I am also fixing cases where ON conditions are more complex.